cfme/tests/physical_infrastructure/api/test_physical_server_api.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils import error
+from cfme.utils.rest import assert_response
+from cfme.utils.wait import wait_for
+from manageiq_client.api import APIException
+
+pytestmark = [
+    pytest.mark.tier(3),
+    pytest.mark.provider([LenovoProvider], scope='module')
+]
+
+# The max time that a single test has to perform the entire cycle of action
+# in seconds (5 min)
+TIMEOUT = 300
+
+# The time interval to perform a verification of state change
+# in seconds (1 min)
+DELAY = 60
+
+
+@pytest.fixture(scope="module")
+def physical_server(setup_provider_modscope, appliance):
+    physical_server = appliance.rest_api.collections.physical_servers[0]
+    return physical_server
+
+
+@pytest.fixture(scope="module")
+def unexistent_physical_server(setup_provider_modscope, appliance):
+    return appliance.rest_api.get_entity('physical_servers', 999999)
+
+
+def do_post_physical_server(params, appliance, physical_server):
+    """
+    Do post againist a single physical server
+    """
+    appliance.rest_api.post(physical_server.href, **params)
+    return appliance.rest_api.response.ok
+
+
+def perform_physical_server_action(action, appliance, physical_server):
+    """
+    Perform an action againist a single physical server
+    """
+    payload = {
+        "action" : action
+    }
+    return do_post_physical_server(payload, appliance, physical_server)
+
+
+def get_server_attr(attr_name, appliance, provider, physical_server):
+    provider.refresh_provider_relationships()
+    physical_server.reload()
+    return physical_server[attr_name]
+
+
+def verify_attribute(server, attribute, value):
+    server.reload()
+    return server[attribute] == value
+
+
+def test_get_physical_server(physical_server, appliance):
+    existent_server = appliance.rest_api.get_entity('physical_servers', physical_server.id)
+    assert existent_server is not None
+
+
+def test_get_nonexistent_physical_server(unexistent_physical_server, appliance):
+    with pytest.raises(APIException) as exception:
+      unexistent_physical_server.name
+    assert exception.match("Couldn't find PhysicalServer with")
+    assert appliance.rest_api.response.status_code == 404
+
+
+def test_server_invalid_action(physical_server, appliance):
+    payload = {
+        "action" : "invalid_action"
+    }
+    with error.expected('Api::BadRequestError'):
+        appliance.rest_api.post(physical_server.href, **payload)
+
+
+def test_refresh_physical_server(appliance, physical_server):
+    assert perform_physical_server_action("refresh", appliance, physical_server)
+
+
+@pytest.mark.parametrize("action, desired_state",[
+    ("power_off", "off"),
+    ("power_on", "on"),
+    ("power_off_now", "off"),
+    ("restart", "on"),
+    ("restart_now", "on")
+])
+def test_server_actions(physical_server, appliance, provider, action, desired_state):
+    assert perform_physical_server_action(action, appliance, physical_server)
```

Comment:
REQUIRED: you can use ``resource.action.action_name()`` in our python API client. Please replace the ``perform_physical_server_action`` calls. In this case it would be smtg like
```python
assert getattr(physical_server.action, action)()
```
This way you can test that the action works AND that it is listed in available actions.

Reply:
Great!